.github/workflows/codeql-analysis.yml

```diff
@@ -34,15 +34,25 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v3
 
+    - run: sed -En 's/^go (.*)$/GO_VERSION=\1/p' go.mod >> $GITHUB_ENV
+      if: ${{ matrix.language == 'go' }}
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+      if: ${{ matrix.language == 'go' }}
+
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v2
       with:
         languages: ${{ matrix.language }}
+        debug: true
+        trap-caching: true
```

Comment:
Why was this added? In any case, I believe this will become the default shortly, so let's remove it. See https://github.com/github/codeql-action/blob/main/CHANGELOG.md#unreleased.